### PR TITLE
chore: add packages' static components from templates

### PR DESCRIPTION
**Pull request by AllanFly120**
Generate `LICENSE`, `.gitignore`, `.npmignore`, `tsconfig.*.json`, `READEME.md` in the AWS client codegen. Once generated for the first time, these files would be overwritten by codegen  again. These entities will be maintained in-place instead of codegen. 

Related: https://github.com/awslabs/smithy-typescript/pull/56

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mtdowling** [comment]
I think the static files and copying them over into code generated clients should happen in the Java codegen.

**Comment by AllanFly120** [comment]
@mtdowling moved static files generation to the code gen.

File: codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsPackageFixturesGeneratorIntegration.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import java.util.Calendar;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.typescript.codegen.TypeScriptSettings;
+import software.amazon.smithy.typescript.codegen.TypeScriptWriter;
+import software.amazon.smithy.typescript.codegen.integration.TypeScriptIntegration;
+import software.amazon.smithy.utils.IoUtils;
+
```
Comment (mtdowling) [inline codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsPackageFixturesGeneratorIntegration.java]: Extra newline

File: codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsPackageFixturesGeneratorIntegration.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    @Override
+    public void writeAdditionalFiles(
+            TypeScriptSettings settings,
+            Model model,
+            SymbolProvider symbolProvider,
+            BiConsumer<String, Consumer<TypeScriptWriter>> writerFactory
+    ) {
+        writerFactory.accept(".gitignore", writer -> {
+            String resource =  IoUtils.readUtf8Resource(
+                    getClass().getClassLoader(), "software/amazon/smithy/aws/typescript/codegen/gitignore");
```
Comment (mtdowling) [inline codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsPackageFixturesGeneratorIntegration.java]: You should be able to just say "gitignore" and leave out the rest of the path

File: codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsPackageFixturesGeneratorIntegration.java (hunk 1 above)
Comment (AllanFly120) [reply codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsPackageFixturesGeneratorIntegration.java]: Resolved

File: codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsPackageFixturesGeneratorIntegration.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.typescript.codegen.TypeScriptSettings;
+import software.amazon.smithy.typescript.codegen.TypeScriptWriter;
+import software.amazon.smithy.typescript.codegen.integration.TypeScriptIntegration;
+import software.amazon.smithy.utils.IoUtils;
+
+public class AwsPackageFixturesGeneratorIntegration implements TypeScriptIntegration {
```
Comment (mtdowling) [inline codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsPackageFixturesGeneratorIntegration.java]: minor: make this final

File: codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsPackageFixturesGeneratorIntegration.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            writer.write(resource);
+        });
+        writerFactory.accept("LICENCE", writer -> {
+            String resource =  IoUtils.readUtf8Resource(getClass(), "LICENCE.template");
+            resource = resource.replace("${year}", Integer.toString(Calendar.getInstance().get(Calendar.YEAR)));
+            writer.write(resource);
+        });
+        writerFactory.accept("README.md", writer -> {
+            String resource =  IoUtils.readUtf8Resource(getClass(), "README.md.template");
+            resource = resource.replaceAll("\\$\\{packageName\\}", settings.getPackageName());
```
Comment (mtdowling) [inline codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsPackageFixturesGeneratorIntegration.java]: minor: `"\\$\\{packageName\\}"` could be replaced with `Pattern.quote("${packageName}")` to make it more readable.

File: codegen/smithy-aws-typescript-codegen/src/test/java/software/amazon/smithy/aws/typescript/codegen/AwsPackageFixturesGeneratorIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import software.amazon.smithy.build.MockManifest;
+import software.amazon.smithy.build.PluginContext;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.typescript.codegen.TypeScriptCodegenPlugin;
+
+public class AwsPackageFixturesGeneratorIntegrationTest {
```
Comment (mtdowling) [inline codegen/smithy-aws-typescript-codegen/src/test/java/software/amazon/smithy/aws/typescript/codegen/AwsPackageFixturesGeneratorIntegrationTest.java]: Sweet!

**Pull request merged**